core/src/banking_stage/consume_worker.rs

```diff
@@ -130,16 +133,22 @@ impl<Tx: TransactionWithMeta> ConsumeWorker<Tx> {
         Ok(())
     }
 
-    /// Try to get a bank for consuming.
-    fn get_consume_bank(&self) -> Option<Arc<Bank>> {
-        self.leader_bank_notifier
-            .get_or_wait_for_in_progress(Duration::from_millis(50))
-            .upgrade()
+    /// Get the current poh working bank with a timeout.
+    fn get_working_bank_with_timeout(&self) -> Option<Arc<Bank>> {
+        const TIMEOUT: Duration = Duration::from_millis(50);
```

Comment:
kept the timeout the same in this PR. I think we may want to remove this in the near future.

Reply:
for my own understanding since this is the first thing that jumped out: when/why
did we need such a long timeout?